content/courses/program-security/reinitialization-attacks.md

```diff
@@ -102,302 +113,280 @@ pub mod initialization_secure {
     use super::*;
 
     pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
-        let user = &mut ctx.accounts.user;
-
+        let mut user = User::try_from_slice(&ctx.accounts.user.data.borrow()).unwrap();
```

Comment:
Is this really necessary? The `&mut ctx.accounts.user;` seems a lot simpler, it looks like we're borrowing twice here, and it seems odd that Anchor would get more complicated over time rather than less.